### ISSUE TITLE
[Security] Method "GuardAuthenticationListener::setSessionAuthenticationStrategy()" does not exist

### ISSUE THREAD

**Issue by derrabus** [symfony/symfony#27577]
**Symfony version(s) affected**: 2.8.42-dev, 3.4.12-dev

**Description**  
Applications configuring a guard authenticator exit with an exception after upgrading to the HEAD of the 2.8/3.4 branch.

```
In AbstractRecursivePass.php line 162:

  [Symfony\Component\DependencyInjection\Exception\RuntimeException]
  Invalid service "security.authentication.listener.guard.main": method "Symfony\Component\Securit
  y\Guard\Firewall\GuardAuthenticationListener::setSessionAuthenticationStrategy()" does not exist
  .
```

**How to reproduce**  
Just configure any guard authenticator on a new Symfony project.

* Reproducer 3.4: https://github.com/derrabus/symfony-standard/tree/3.4-broken-guard
* Reproducer 2.8: https://github.com/derrabus/symfony-standard/tree/2.8-broken-guard

**Additional context**  
<img width="1034" alt="bildschirmfoto 2018-06-11 um 18 44 04" src="https://user-images.githubusercontent.com/1506493/41245137-7ab8c652-6da7-11e8-866f-bd544bb68ddd.png">

<img width="979" alt="bildschirmfoto 2018-06-11 um 19 53 58" src="https://user-images.githubusercontent.com/1506493/41248415-4d187a76-6db1-11e8-9ccb-84eb26f8ab55.png">

If you downgrade `symfony/symfony` to 2.8.41/3.4.11, everything's fine again.

**Comment by derrabus** [comment]
Could this be related to #27452? /cc @weaverryan

**Comment by weaverryan** [comment]
My fault for sure - I'm working on it :)

**Comment by chalasr** [comment]
Thanks for testing dev branches @derrabus :)

**Comment by derrabus** [comment]
@weaverryan I can confirm that the issue is gone on the 2.8 branch. However, I can still reproduce it on 3.4 although #27581 must have been merged there as well by now. Can you have a look, please?

**Comment by chalasr** [comment]
@derrabus fixed in 13e983a127b9a55aa7ffebe8da3ff88958f712ef, thanks!

**Comment by derrabus** [comment]
Wow, that was fast, @chalasr! 3.4 seems to be working now as well. Thanks a lot!
